compiler/src/iree/compiler/Codegen/Common/GPU/GPUReduceBankConflicts.cpp

```diff
@@ -74,6 +74,77 @@ static void padAlloc(MLIRContext *context, memref::AllocOp allocOp,
   rewriter.eraseOp(allocOp);
 }
 
+static int64_t computeSharedMemoryUsage(mlir::FunctionOpInterface funcOp) {
+  int64_t totalSharedMemory = 0;
+
+  auto walkResult = funcOp.walk([&](memref::AllocOp allocOp) -> WalkResult {
+    if (!hasSharedMemoryAddressSpace(allocOp.getType())) {
+      return WalkResult::advance();
+    }
+
+    if (!allocOp.getType().hasStaticShape()) {
+      return WalkResult::interrupt();
+    }
+
+    MemRefType allocType = llvm::cast<MemRefType>(allocOp.getType());
+    unsigned bitWidth =
+        allocType.getElementType().isIndex()
+            ? 64 // IREE's default bitWidth for indexes
+            : IREE::Util::getTypeBitWidth(allocType.getElementType());
+    if (allocOp.getAlignment()) {
```

Comment:
You have to do this on allocSizeBytes after you have calculated it, the idea is that lets say you have 7 bytes of data, but aligned to 64 bit, you will use (ceil(7/64)*64)/8 => 64/8 => 8 bytes

This calculation should not include bitWidth at all.

Reply:
I think my math is wrong, maybe you want ceil(7x8/64)*64)/8

Reply:
so you cold just use the alignment in Bytes and do ceil(7/8)*8

Reply:
No, how you have it is fine. Sorry I said too many different things. So the idea of alignment is that you reserve the bytes in that alignment multiple. The alignment is provided in bytes. So let's say your allocs are 1,7,9,17 bytes and your alignment is 8 bytes, you will reserve 8,8,16,24 bytes. So what you need to do is remove what you have for alignment currently and just do 
```
if(alignment)
  allocSizeByes = ceil(allocSizeBytes/alignment) * alignment;
```
Please let me know if there are further questions.